tests/filtertransformers/test_mongo.py

```diff
@@ -341,8 +342,170 @@ def test_not_implemented(self):
+
+    def test_list_length_aliases(self):
+        from optimade.server.mappers import StructureMapper
+
+        transformer = MongoTransformer(mapper=StructureMapper())
+        parser = LarkParser(version=self.version, variant=self.variant)
+
+        self.assertEqual(
+            transformer.transform(parser.parse("elements LENGTH 3")), {"nelements": 3}
+        )
+
+        self.assertEqual(
+            transformer.transform(
+                parser.parse('elements HAS "Li" AND elements LENGTH = 3')
+            ),
+            {"$and": [{"elements": {"$in": ["Li"]}}, {"nelements": 3}]},
+        )
+
+        self.assertEqual(
+            transformer.transform(parser.parse("elements LENGTH > 3")),
+            {"nelements": {"$gt": 3}},
+        )
+        self.assertEqual(
+            transformer.transform(parser.parse("elements LENGTH < 3")),
+            {"nelements": {"$lt": 3}},
+        )
+        self.assertEqual(
+            transformer.transform(parser.parse("elements LENGTH = 3")), {"nelements": 3}
+        )
+        self.assertEqual(
+            transformer.transform(parser.parse("cartesian_site_positions LENGTH <= 3")),
+            {"nsites": {"$lte": 3}},
+        )
+        self.assertEqual(
+            transformer.transform(parser.parse("cartesian_site_positions LENGTH >= 3")),
+            {"nsites": {"$gte": 3}},
+        )
+
+    def test_unaliased_length_operator(self):
+        transformer = MongoTransformer()
+        parser = LarkParser(version=self.version, variant=self.variant)
+        self.assertEqual(
+            transformer.transform(parser.parse("cartesian_site_positions LENGTH <= 3")),
+            {"cartesian_site_positions.4": {"$exists": False}},
+        )
+        self.assertEqual(
+            transformer.transform(parser.parse("cartesian_site_positions LENGTH < 3")),
+            {"cartesian_site_positions.3": {"$exists": False}},
+        )
+        self.assertEqual(
+            transformer.transform(parser.parse("cartesian_site_positions LENGTH 3")),
+            {"cartesian_site_positions": {"$size": 3}},
+        )
+        self.assertEqual(
+            transformer.transform(
+                parser.parse("cartesian_site_positions LENGTH >= 10")
+            ),
+            {"cartesian_site_positions.10": {"$exists": True}},
+        )
+
+        self.assertEqual(
+            transformer.transform(parser.parse("cartesian_site_positions LENGTH > 10")),
+            {"cartesian_site_positions.11": {"$exists": True}},
+        )
+
+    def test_aliased_length_operator(self):
+        from optimade.server.mappers import StructureMapper
+
+        class MyMapper(StructureMapper):
+            ALIASES = (("elements", "my_elements"), ("nelements", "nelem"))
+            LENGTH_ALIASES = (
+                ("chemsys", "nelements"),
+                ("cartesian_site_positions", "nsites"),
+                ("elements", "nelements"),
+            )
+            PROVIDER_FIELDS = ("chemsys",)
```

Comment:
Is this the main reason for introducing `PROVIDER_FIELDS`?

Reply:
Yes, I think it makes sense that provider fields has parity with the other aliases, so we should allow people to define it at the mapper level too, so a particular implementation that always defines a set of provider fields does not have to rely on them being defined in the config. This means you can then have a self-contained mapper that can define length aliases on provider fields directly too. This one is obviously just a toy example!

Reply:
Fair enough.
And sure, this is a toy example :)
Was just fishing for an explicit justification, not a particularly well-argumented one, since I saw this was an additional variable added in this PR ;)